### ISSUE TITLE
Custom message editor

### ISSUE THREAD

**Issue by SethBurkart123** [BetterSEQTA/BetterSEQTA-Plus#11]

**Comment by WizardEels** [comment]
Hey! wdym my this?

**Comment by MEGA-Dawg68** [comment]
I think (No offense) that if you got an idea, put a description. it doesnt have to be long just the gist of what you are trying to do

**Comment by SethBurkart123** [comment]
Basically just a better text editor for direct messages. An idea might be to use something like https://blocknotejs.org/

**Comment by ar-cyber** [comment]
The SEQTA one is already pretty good. It may just need a little more features.

**Comment by SethBurkart123** [comment]
> The SEQTA one is already pretty good. It may just need a little more features.

I mean 😢  I just find it a bit annoying to use (and ugly) and there are a bunch of features like having image uploads embed directly into the message.

**Comment by ar-cyber** [comment]
How's the progress going on this one?

**Comment by SethBurkart123** [comment]
> How's the progress going on this one?

this one's turned out to be a bit messy so I've delayed it a little

**Comment by 97-42** [comment]
If we are to use https://blocknotejs.org/ we should find a way to cover costs. I'm not suggesting implemting ads but we dont want to be spending money if there is no way of making it back.

**Comment by SethBurkart123** [comment]
> If we are to use https://blocknotejs.org/ we should find a way to cover costs. I'm not suggesting implemting ads but we dont want to be spending money if there is no way of making it back.

Don't worry, we don't actually need to pay for it. It's just got more detailed examples- it's also for proper companies with support etc.

**Comment by ar-cyber** [comment]
How would the messages be sent? The issue is that SEQTA (as a whole) probably cannot render many of the things in other text editors. Would it be sent as HTML?

**Comment by ar-cyber** [comment]
> > The SEQTA one is already pretty good. It may just need a little more features.
> 
> I mean 😢 I just find it a bit annoying to use (and ugly) and there are a bunch of features like having image uploads embed directly into the message.

don't even get me started on links.

**Comment by SethBurkart123** [comment]
> How would the messages be sent? The issue is that SEQTA (as a whole) probably cannot render many of the things in other text editors. Would it be sent as HTML?

Well seqta direct messages are just html pages, so you can actually just send html with a bunch of css to make it work perfectly. It also means that I can make it fully support betterseqta+ dark mode as I add the `.dark` class to the body element of it...

**Comment by ar-cyber** [comment]
> > How would the messages be sent? The issue is that SEQTA (as a whole) probably cannot render many of the things in other text editors. Would it be sent as HTML?
> 
> Well seqta direct messages are just html pages, so you can actually just send html with a bunch of css to make it work perfectly. It also means that I can make it fully support betterseqta+ dark mode as I add the `.dark` class to the body element of it...

which would mean the body of the text in the custom message editor should be sent as html?

**Comment by SethBurkart123** [comment]
> > > How would the messages be sent? The issue is that SEQTA (as a whole) probably cannot render many of the things in other text editors. Would it be sent as HTML?
> > 
> > 
> > Well seqta direct messages are just html pages, so you can actually just send html with a bunch of css to make it work perfectly. It also means that I can make it fully support betterseqta+ dark mode as I add the `.dark` class to the body element of it...
> 
> which would mean the body of the text in the custom message editor should be sent as html?

yes exactly! with inlined css styles using `<style></style>` tags. But one annoying thing is seqta adds their own styles on top too, so we'd have to override those specifically.

**Comment by ar-cyber** [comment]
> > > > How would the messages be sent? The issue is that SEQTA (as a whole) probably cannot render many of the things in other text editors. Would it be sent as HTML?
> > > 
> > > 
> > > Well seqta direct messages are just html pages, so you can actually just send html with a bunch of css to make it work perfectly. It also means that I can make it fully support betterseqta+ dark mode as I add the `.dark` class to the body element of it...
> > 
> > 
> > which would mean the body of the text in the custom message editor should be sent as html?
> 
> yes exactly! with inlined css styles using `<style></style>` tags. But one annoying thing is seqta adds their own styles on top too, so we'd have to override those specifically.

couldn't we just override and display a custom html page over the seqta stuff

**Comment by SethBurkart123** [comment]
> couldn't we just override and display a custom html page over the seqta stuff

I mean the big thing is keeping it still compatible so that people that don't use BS+ can still enjoy the nice messages if you get what I mean.

**Comment by Crazypersonalph** [comment]
> > couldn't we just override and display a custom html page over the seqta stuff
> 
> I mean the big thing is keeping it still compatible so that people that don't use BS+ can still enjoy the nice messages if you get what I mean.

What a setup with embeds or iframes? By default, SEQTA opens links in the message themselves (annoyingly enough), but we could leverage this to load a custom message. Otherwise, we could also use the inline css.

**Comment by SethBurkart123** [comment]
> What about a setup with embeds or iframes? By default, SEQTA opens links in the message themselves (annoyingly enough), but we could leverage this to load a custom message. Otherwise, we could also use the inline css.

Yes, that's actually because the messages are iframes..

But custom messages are totally doable, they are actually sent as HTML so you can basically do anything you want with css and html. Take for example this working css only stack game I sent to a friend:

<img width="258" alt="Image" src="https://github.com/user-attachments/assets/918b67a2-1e06-46f1-a24c-47d44c61700c" />

**Comment by ar-cyber** [comment]
it (kinda) starts:

![Image](https://github.com/user-attachments/assets/3c3ddfb8-a053-4c89-976d-4ad16ec63a46)

**Comment by NIDNHU** [comment]
> > If we are to use https://blocknotejs.org/ we should find a way to cover costs. I'm not suggesting implemting ads but we dont want to be spending money if there is no way of making it back.
> 
> Don't worry, we don't actually need to pay for it. It's just got more detailed examples- it's also for proper companies with support etc.

i know I'm a bit late, but because it's open source, couldn't you implement it directly?

**Comment by ar-cyber** [comment]
Someone push this to in progress.

**Comment by ar-cyber** [comment]
have a working demo of this:

![Image](https://github.com/user-attachments/assets/ee0f6965-5818-4887-bcff-e8308d765ed1)

**Comment by ar-cyber** [comment]
someone close this as it is an unofficial plugin now
